util/topgen/templates/chiplevel.sv.tpl

```diff
@@ -1069,7 +1100,7 @@ module chip_${top["name"]}_${target["name"]} #(
 // Also need to add AST simulation and FPGA emulation models for things like entropy source -
 // otherwise Verilator / FPGA will hang.
   top_${top["name"]} #(
-% if target["name"] == "cw310":
+% if target["name"] in ["cw310", "cw340"]:
```

Comment:
Eventually, we will want to add a separate branch for the `CW340/CW341` here to enable also KMAC masking on FPGA. For this, both `KeymgrKmacEnMasking` and `KmacEnMasking` need to be set to 1. But I recommend doing this as a follow-up. It will have a big impact on utilization, timing and maybe also build time.

Reply:
With those enabled, it consumed another 3% of the available logic, and timing was not materially affected. The build time did go up another 15-20% locally (about 6-8 minutes) for the 30 MHz SYS / 33.333 MHz IO build.